cmd/cmd.go

```diff
@@ -1,93 +1,104 @@
 
 	flag.Parse()
 
 	if *printVersion {
-		fmt.Println(Version)
+		//fmt.Println(Version)
 		os.Exit(0)
 	}
 
-	if *functionCmd == "" {
-		fmt.Println("Required command to execute")
+	if *activityName == "" {
+		fmt.Println("activityname is required")
 		os.Exit(1)
 	}
-
-	var job runner.BatchJob
-	if *localRun {
-		// TODO: allow CLI to set fake job-ids (or inputs)
-		job = runner.NewMockBatchJob([]string{})
-	} else {
-		j, err := runner.NewBatchJobFromEnv()
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		}
-		job = j
-	}
-
-	if *resultsLocation == "" {
-		fmt.Println("-results-location can not be an empty string")
+	if *activityCmd == "" {
+		fmt.Println("cmd is required")
 		os.Exit(1)
 	}
-
-	// TODO: figure out a better way to switch between prod/dev resultsLocations
-	// currently using env var since that easily works across docker containers
-	// which specify `CMD [..]` or `ENTRYPOINT [...]`
-	if strings.Contains(os.Getenv("AWS_BATCH_JQ_NAME"), "production") {
-		*resultsLocation = "workflow-results"
-	}
-
-	store, err := initalizeStore(*localRun, *resultsLocation)
-	if err != nil {
-		fmt.Println(err)
+	if *sfnRegion == "" {
+		fmt.Println("region is required")
 		os.Exit(1)
 	}
 
-	taskRunner, err := runner.NewTaskRunner(*functionCmd, flag.Args(), job, store)
+	ctx := context.Background()
+
+	// register the activity with AWS (it might already exist, which is ok)
+	sfnapi := sfn.New(session.New(), aws.NewConfig().WithRegion(*sfnRegion))
+	createOutput, err := sfnapi.CreateActivityWithContext(ctx, &sfn.CreateActivityInput{
+		Name: activityName,
+	})
 	if err != nil {
-		fmt.Println(err)
+		fmt.Printf("error creating activity: %s\n", err)
 		os.Exit(1)
+	} else {
+		fmt.Printf("running as activity: %s\n", *createOutput.ActivityArn)
 	}
 
-	if err := taskRunner.Process(); err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-	// TODO: fix this with a chan and proper buffer handling
-	// wait a sec for the output buffer to process
+	// run getactivitytask and get some work
+	// getactivitytask itself claims to initiate a polling loop, but wrap it in a polling loop of our own
+	// since it seems to return every minute or so with a nil error and empty output
+	ticker := time.NewTicker(5 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			fmt.Println("terminating GetActivityTask polling loop")
+			break
+		case <-ticker.C:
+			getATOutput, err := sfnapi.GetActivityTaskWithContext(ctx, &sfn.GetActivityTaskInput{
+				ActivityArn: createOutput.ActivityArn,
+				// WorkerName: "TODO: something useful about this process: host? pid?",
+			})
+			if err != nil {
+				fmt.Printf("error getting activity task: %s\n", err)
+				os.Exit(1)
+			}
+			if getATOutput.TaskToken == nil {
+				fmt.Println("nil task token starting GetActivityTask again")
+				continue
+			}
+			fmt.Println("got activity task", getATOutput)
+			activityTaskInput := getATOutput.Input
```

Comment:
TODO: need to figure out our `Input` constraints we want to enforce in workflow-manager and maybe check them here again.